src/execution/operator/persistent/buffered_csv_reader.cpp

```diff
@@ -455,6 +550,61 @@ vector<LogicalType> BufferedCSVReader::SniffCSV(vector<LogicalType> requested_ty
 					const auto &sql_type = col_type_candidates.back();
 					// try cast from string to sql_type
 					auto dummy_val = parse_chunk.GetValue(col, row);
+					// try formatting for date types if the user did not specify one and it starts with numeric values.
+					string separator;
+					if (has_format_candidates.count(sql_type.id()) && !original_options.has_format[sql_type.id()] &&
+					    StartsWithNumericDate(separator, dummy_val.str_value)) {
+						// generate date format candidates the first time through
+						auto &type_format_candidates = format_candidates[sql_type.id()];
+						const auto had_format_candidates = has_format_candidates[sql_type.id()];
+						if (!has_format_candidates[sql_type.id()]) {
+							has_format_candidates[sql_type.id()] = true;
+							// order by preference
+							for (const auto &t : format_template_candidates[sql_type.id()]) {
+								const auto format_string = GenerateDateFormat(separator, t);
+								// don't parse ISO 8601
+								if (format_string.find("%Y-%m-%d") == string::npos)
+									type_format_candidates.emplace_back(format_string);
+							}
+
+							//	initialise the first candidate
+							options.has_format[sql_type.id()] = true;
+							//	all formats are constructed to be valid
+							SetDateFormat(type_format_candidates.back(), sql_type.id());
+						}
+						// check all formats and keep the first one that works
+						int32_t date_parts[6];
```

Comment:
This should be [7], not [6] :)

Reply:
Another victory for strong typing... We should change the argument to Parse to be a struct that has all the output variables. But that is a problem for another day.

Reply:
Indeed, good point.